src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -188,20 +187,6 @@ class Peer(
     val swapInAddress: String = nodeParams.keyManager.swapInOnChainWallet.address.also { swapInWallet.addAddress(it) }
 
     init {
-        launch {
-            watcher.client.notifications.filterIsInstance<HeaderSubscriptionResponse>()
```

Comment:
There was a race condition here between establishing the connection to electrum (which happens outside of the peer), and subscribing to `HeaderSubscriptionResponse`.

If the connection to electrum is very fast, tje peer will miss the initial `HeaderSubscriptionResponse` and then have to wait for the next block.

Same race condition applies for feerates.